From 4d97d70aeca42761c3fb1a74af1f315389d77908 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Fri, 23 Dec 2022 12:56:56 +0200
Subject: [PATCH] [SW-113135] ppboot: change sequence for sram extension

Currently changing SRAM extension address for ARC1 required for onix
platform is done after preboot delay routine. Starting gaudi3, ppboot
make use of SRAM for bootinfo operations in ppboot, so this patch change
extension setting sequence accordingly.

Change-Id: I5780f6a21900fc29dd509e880af7007f4cc1fafe
---

diff --git a/app/ppboot/ppboot.c b/app/ppboot/ppboot.c
index 3f62e54..73d9f31 100644
--- a/app/ppboot/ppboot.c
+++ b/app/ppboot/ppboot.c
@@ -312,6 +312,8 @@
 	device_init();
 	/* Set ARCs to privileged mode */
 	ppboot_secure_arc();
+	sram_set_extension();
+
 	/*
 	 * Note: we assume CPU frequency is not changed while CPU is running,
 	 * so the udelay init is done once before we start.
@@ -337,7 +339,6 @@
 	hl_plat_init();
 
 	preboot_delay();
-	sram_set_extension();
 
 	hl_sram_self_repair();
 
